migrations/versions/7590a700f9bc_.py

```diff
@@ -0,0 +1,201 @@
+        "Report": ["%wes_version.txt"],
+    },
+    "RNA": {
+        "Source": ["%rna%reads_%.bam", "%rna%r1_%.fastq.gz", "%rna%r2_%.fastq.gz"],
+        "Alignment": [
+            "%sorted.bam",
+            "%sorted.bam.bai",
+            "%sorted.bam.stat.txt",
+            "%downsampling.bam",
+            "%downsampling.bam.bai",
+        ],
+        "Quality": [
+            "%downsampling_housekeeping.bam",
+            "%downsampling_housekeeping.bam.bai",
+            "%read_distrib.txt",
+            "%tin_score.txt",
+            "%tin_score.summary.txt",
+        ],
+        "Gene Quantification": [
+            "%quant.sf",
+            "%transcriptome.bam.log",
+            "%aux_info_ambig_info.tsv",
+            "%aux_info_expected_bias.gz",
+            "%aux_info_fld.gz",
+            "%aux_info_meta_info.json",
+            "%aux_info_observed_bias.gz",
+            "%aux_info_observed_bias_3p.gz",
+            "%cmd_info.json",
+            "%salmon_quant.log",
+        ],
+    },
+    "mIF": {
+        "Source Images": [
+            "%composite_image.tif",
+            "%component_data.tif",
+            "%multispectral.im3",
+        ],
+        "Analysis Images": ["%binary_seg_maps.tif", "%phenotype_map.tif"],
+        "Analysis Data": [
+            "%score_data_%.txt",
+            "%cell_seg_data.txt",
+            "%cell_seg_data_summary.txt",
+        ],
+    },
+    "Olink": {"All Olink Files": ["%/olink%"]},
+    "IHC": {"All IHC Files": ["%/ihc%"]},
+}
+
+clinical_facets = {
+    "Participants Info": ["%participants.csv"],
+    "Samples Info": ["%samples.csv"],
+}
+
+
+facets = {"Assay Type": assay_facets, "Clinical Type": clinical_facets}
+
+
+def list_facet_paths():
+    paths = []
+    for assay_name, subfacets in assay_facets.items():
+        for subfacet in subfacets.keys():
+            paths.append(["Assay Type", assay_name, subfacet])
+
+    for clinical_type in clinical_facets.keys():
+        paths.append(["Clinical Type", clinical_type])
+
+    return paths
+
+
+def get_facets_for_path(object_url_like, path):
+    try:
+        assert len(path) in (2, 3)
+        facet_config = facets
+        for key in path:
+            facet_config = facet_config[key]
+        assert isinstance(facet_config, list)
+    except Exception as e:
+        raise ValueError(f"no facet for path {path}")
+
+    clauses = [object_url_like(clause) for clause in facet_config]
+    return clauses
+
+
+def build_facet_group_for_existing_files():
+    """
+    Builds a sqlalchemy `case` statement for figuring out which 
+    facet group a particular record belongs to.
+
+    This will build facet groups like "Assay Type|mIF|Source Images"
+    """
+    cases = []
+    for path in list_facet_paths():
+        object_url_like = lambda exp: sa.text(f"object_url LIKE '{exp}'")
+        path_facets = sa.or_(*get_facets_for_path(object_url_like, path))
+        facet_group = "|".join(path)
```

Comment:
I think would be better to have more granular facet groups, eg `Assay|RNA|Source|rna_reads_.bam` etc.